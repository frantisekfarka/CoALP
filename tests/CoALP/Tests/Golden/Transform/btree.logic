bit(0).
bit(1).
btree(empty).
btree(tree(L,X,R)) :- btree(L), bit(X), btree(R).
