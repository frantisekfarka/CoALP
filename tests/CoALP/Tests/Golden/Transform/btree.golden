bit(0, transform-func-1).
bit(1, transform-func-2).
btree(empty, transform-func-3).
btree(tree(V_2i, V_3i, V_4i), transform-func-4(V_5i, V_6i, V_7i)) :- btree(V_2i, V_5i), bit(V_3i, V_6i), btree(V_4i, V_7i).
